### DISCUSSION TITLE
Display Rich Content

### DISCUSSION THREAD

**Discussion by RajatGurnani** [Chatterino/chatterino2#4048] [Ideas]
A support for **displaying rich content embed** in the chat window for the Chatterino desktop app would be great, something similar to what FrankerFaceZ browser extension does:- 

**Before**
![image](https://user-images.githubusercontent.com/38850724/194122945-680060fc-6fed-47e6-9534-e57a7c33f209.png)
**After**
![image](https://user-images.githubusercontent.com/38850724/194122094-c23f426a-0483-4fe3-a4eb-7a1750985bff.png)

**FrankerFaceZ options for reference-**
![image](https://user-images.githubusercontent.com/38850724/194123672-a2c0b85f-dcc2-4643-a16e-38dd1b97344b.png)

**Comment by Felanbird** [comment]
![image](https://user-images.githubusercontent.com/41973452/194124766-5732426e-8a52-4a6d-b34b-9b13a0bc15de.png)
We already have this feature

**Reply by RajatGurnani** [reply]
I meant that it can be embedded in the chat itself rather than making it hover only. Sorry, should've clarified that in the post.

**Reply by Felanbird** [reply]
> I meant that it can be embedded in the chat itself rather than making it hover only. Sorry, should've clarified that in the post.

I did realize u wanted embeds, but due to your blank before link, I was unsure if you knew about the hover implementation.

**Reply by RajatGurnani** [reply]
Yeah that blank was due to the chat window size.
Is there any implementation in the works for it?

**Reply by Felanbird** [reply]
There is not, most effort towards this sort of thing goes directly into the existing hover. 
Like chatterino/api#366 making sure thumbnails respect rotation metadata

**Comment by pajlada** [comment]
A theoretical v2 implementation of chatterino/api would be transformed to a rich JSON API instead and it'd be up to the client for how to render it, meaning we could theoretically render it in ways other than a tooltip
